### ISSUE TITLE
azurerm_frontdoor_custom_https_configuration tries to update resource_group_name

### ISSUE THREAD

**Issue by rscottwatson** [terraform-providers/terraform-provider-azurerm#10943]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Affected Resource(s)

* `azurerm_2.50.0 and 2.49.0`

### Terraform Configuration Files

```hcl
resource "azurerm_frontdoor_custom_https_configuration" "custom_https_1" {
    frontend_endpoint_id              = azurerm_frontdoor.main.frontend_endpoints[local.HOST_SHORTNAME]
    custom_https_provisioning_enabled = true
    #
    # this is not a parameter but if we don't supply it each time we run a plan
    # it will try to set this value to null and cause the apply to fail saying that 
    # Message="That action isn’t allowed in this profile."
    # however when we include it we get a depreacted warning that it is going to be removed in the next release.
    #
   # resource_group_name               = var.resource_group_name
    custom_https_configuration {
        certificate_source   = "FrontDoor"
    }
}
```

### Debug Output
https://gist.github.com/rscottwatson/b24343ff1c086124f9a582b0b5ec426a

### Expected Behaviour
It should have detected no changes and not tried to set the resource group to null

### Actual Behaviour
It tries to set it to null and fails

### Steps to Reproduce

Create a azurerm_frontdoor_custom_https_configuration
apply it
run apply again

### References

* #0000

**Comment by bcochofel** [comment]
This issue still happens with AzuremRM v2.54.0

**Comment by oliwer** [comment]
I did not get this error using azurerm v2.60.0. I followed the [2.58.0 migration tutorial](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/guides/2.58.0-frontdoor-upgrade-guide).

**Comment by tombuildsstuff** [comment]
Closing as confirmed as fixed in v2.60 by @oliwer above
